File: 4.py
zeichenkette = ["123", "1234", "12345", "1234566", "1234567"]
i = 0
maxLen = 0

while i < len(zeichenkette):
    if len(zeichenkette[i]) > maxLen:
        maxLen = len(zeichenkette[i])
    i += 1

i = 0 

laengste = []
while i < len(zeichenkette):
    if len(zeichenkette[i]) == maxLen:
        laengste.append(zeichenkette[i])
    i += 1

print(f"Die längste Zeichenkette: {laengste}")

#def laengste_zeichenketten(liste):
#    maxLen = maxLen(len(i) for i in liste)
#    return [i for i in liste if len(i) == maxLen]

#laengste = laengste_zeichenketten(zeichenkette)

File: 4.js
let zeichenkette = ["123", "1234", "12345", "1234566", "1234567"];
let i = 0;
let maxLen = 0;


while (i < zeichenkette.length) {
    if (zeichenkette[i].length > maxLen) {
        maxLen = zeichenkette[i].length;
    }
    i++;
}

i = 0;
const laengste = [];

while (i < zeichenkette.length) {
    if (zeichenkette[i].length === maxLen) {
        laengste.push(zeichenkette[i]);
    }
    i++;
}

console.log(`Die längste Zeichenkette: ${JSON.stringify(laengste)}`);